dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -528,19 +707,29 @@ private void launch(final List<String> args) {
         if (args.isEmpty() || containsHelpRequest(args)) {
             launchHelp();
         } else {
-            final String descriptor = optVal(args, "--descriptor", CWL_STRING);
-
-            if (descriptor.equals(CWL_STRING)) {
-                try {
-                    launchCwl(args);
-                } catch (ApiException e) {
-                    exceptionMessage(e, "api error launching workflow", Client.API_ERROR);
-                } catch (IOException e) {
-                    exceptionMessage(e, "io error launching workflow", IO_ERROR);
+            if (args.contains("--local-entry")) {
+                final String descriptor = optVal(args, "--descriptor", null);
+                List<String> argsList = new ArrayList<>();
```

Comment:
This line of code(714), to get the entry file: " reqVal(args, "--entry"); " for checking the content and extension of the file before passing it to launchWdl or launchCwl method. After this is called, the number of args is reduced("--entry <filename> is no longer in the args list after reqVal is called). Therefore, it is copied first into a List of args for entry file checking, and this List will then be passed to launchWdl or launchCwl after file checking.